src/rng.cpp

```diff
@@ -8,18 +8,12 @@
 #define _USE_MATH_DEFINES
 #include <cmath>
 
-long rng( long val1, long val2 )
+template<class T>
+const T& rng(const T& val1, const T& val2)
```

Comment:
This needs to be limited to numeric types as @BevapDin mentioned.
See http://en.cppreference.com/w/cpp/types/enable_if for the feature, and
https://github.com/CleverRaven/Cataclysm-DDA/blob/4589d3bddb94359cdf9c200bcd5fe44d8fcf0481/src/common_types.h#L13 for an example where it is used.